src/framework/pybind_json.hpp

```diff
@@ -384,146 +434,159 @@ py::object AerToPy::from_avg_snap(const AER::AverageSnapshot<T> &avg_snap) {
       snapshots[pair.first.data()] = d;
     }
-    for (auto &pair : result.pershot_cvector_snapshots_) {
+    for (auto &pair : datum.pershot_cvector_snapshots_) {
       py::dict d;
       // string PershotData
-      for (auto &per_pair : pair.second.data())
-        d[per_pair.first.data()] = per_pair.second.data();
+      for (auto &per_pair : pair.second.data()) {
+        py::list l;
+        for (auto &matr : per_pair.second.data())
+          l.append(AerToPy::array_from_vector(matr));
+        d[per_pair.first.data()] = l;
+      }
       snapshots[pair.first.data()] = d;
     }
-    for (auto &pair : result.pershot_cmatrix_snapshots_) {
+    for (auto &pair : datum.pershot_cmatrix_snapshots_) {
       py::dict d;
       // string PershotData
-      for (auto &per_pair : pair.second.data())
-        d[per_pair.first.data()] = per_pair.second.data();
+      for (auto &per_pair : pair.second.data()) {
+        py::list l;
+        for (auto &matr : per_pair.second.data())
+          l.append(AerToPy::array_from_matrix(matr));
+        d[per_pair.first.data()] = l;
+      }
       snapshots[pair.first.data()] = d;
     }
-    for (auto &pair : result.pershot_cmap_snapshots_) {
+    for (auto &pair : datum.pershot_cmap_snapshots_) {
       py::dict d;
       // string PershotData
       for (auto &per_pair : pair.second.data())
         d[per_pair.first.data()] = per_pair.second.data();
       snapshots[pair.first.data()] = d;
     }
-    for (auto &pair : result.pershot_rmap_snapshots_) {
+    for (auto &pair : datum.pershot_rmap_snapshots_) {
       py::dict d;
       // string PershotData
       for (auto &per_pair : pair.second.data())
         d[per_pair.first.data()] = per_pair.second.data();
       snapshots[pair.first.data()] = d;
     }
+
     if ( py::len(snapshots) != 0 )
-        pyresult["snapshots"] = snapshots;
+        pydata["snapshots"] = snapshots;
   }
-  //for (auto item : pyresult)
+  //for (auto item : pydatum)
   //  py::print("    {}:, {}"_s.format(item.first, item.second));
-  return std::move(pyresult);
+  return pydata;
 }
 
-py::object AerToPy::from_exp_result(const AER::ExperimentResult &result) {
-  py::dict pyresult;
+py::object AerToPy::from_experiment(const AER::ExperimentResult &result) {
+  py::dict pyexperiment;
 
-  pyresult["shots"] = result.shots;
-  pyresult["seed_simulator"] = result.seed;
+  pyexperiment["shots"] = result.shots;
+  pyexperiment["seed_simulator"] = result.seed;
 
-  pyresult["data"] = AerToPy::from_exp_data(result.data);
+  pyexperiment["data"] = AerToPy::from_data(result.data);
 
-  pyresult["success"] = (result.status == AER::ExperimentResult::Status::completed);
+  pyexperiment["success"] = (result.status == AER::ExperimentResult::Status::completed);
   switch (result.status) {
     case AER::ExperimentResult::Status::completed:
-      pyresult["status"] = std::string("DONE");
+      pyexperiment["status"] = std::string("DONE");
```

Comment:
in C++14 we have this beauty syntactic sugar construction: `"this is a std::string"s`

Reply:
ha - good catch - I think in the early days I was having trouble with the templating and at one point it was not able to recognize the old const "DONE" char literals as strings which is why I put the manual ctor call in - I'll take out the ctor but I'm hesitant to use the 14-specific feature, just for backwards compatibility (even though it is probably more efficient in some cases)